server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -377,8 +380,20 @@ public void createEntryTable(final String tableName)
                 + "  PRIMARY KEY (id)\n"
                 + ")",
                 tableName, getPayloadType(), getCollation()
-            ),
-            StringUtils.format("CREATE INDEX idx_%1$s_active_created_date ON %1$s(active, created_date)", tableName)
+            )
+        )
+    );
+    createIndex(
+        tableName,
+        "active_created",
+        StringUtils.format("CREATE INDEX idx_%1$s_active_created_date ON %1$s(active, created_date)", tableName)
+    );
+    createIndex(
+        tableName,
+        "datasource_active",
+        StringUtils.format(
+            "CREATE INDEX idx_%1$s_datasource_active_date ON %1$s(datasource, active)",
+            tableName
```

Comment:
Have you considered moving these create index commands to the function `alterEntryTable` below? instead of it being part of the function `createEntryTable`

Reply:
We are creating index along with the createTable method for all meta tables, thus putting it along with Create table. 
Ideally We should migrate other create index statements to this new construct, so that we attempt to create the indexes if they are missing esp older installations.